### ISSUE TITLE
Query Records by empty or non-empty values

### ISSUE THREAD

**Issue by esopian** [balderdashy/waterline#189]
Need a way to get a list of records based on whether they have a property assigned or not and not based on the actual value of the property.  Tried to query with null, empty string, undefined, and false using the MongoDB adapter with no success.

_Example use case:  I have a collection of Invoices with a property of "paidDate".  I want to get all Invoices with no paidDate set (unpaid invoices)._

**Comment by dmarcelino** [comment]
I'd like to resurrect this issue if nobody objects. This seems to be very useful functionality and several people have shown interest both in this thread and in stackoverflow [#27682106](http://stackoverflow.com/questions/27682106/sailsjs-waterline-query-where-not-empty).

**Comment by dmarcelino** [comment]
My proposal is to use a [criteria modifier](https://github.com/balderdashy/waterline-docs/blob/master/query-language.md#criteria-modifiers), `empty`, so one could query for empty like:
* `model.find( field: { empty: true } )`

We also need to consider how this would behave in schemaless DBs where the field may not exist at all. If we want to be more specific we could add another criteria modifier to cover the case where the field doesn't exist at all such as `exists`:
* `model.find( field: { exists: false } )`

Any opinions about this?

**Comment by balnagy** [comment]
+1. Any workaround?

**Comment by sailsbot** [comment]
Thanks for posting, @esopian.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Reopening until someone submits a PR to the roadmap for this feature.

**Comment by CarlosVitorino** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @esopian.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by particlebanana** [comment]
> Reopening until someone submits a PR to the roadmap for this feature.

**Comment by randallmeeker** [comment]
> Reopening until someone submits a PR to the road map for this feature.

It's on the road map.

**Comment by sohail0992** [comment]
Is this functionality implemented or still on the road map
